### PR TITLE
feat: embed testnet faucet in `create-fuels` template

### LINKED ISSUE #2995: Integrate embedded testnet faucet via iframe in `create fuels` template
Right now, users have to open a new tab to be able to use the testnet faucet. They should ideally be able to do this _inside_ the template app itself, using an iframe of some sort.

### PR DESCRIPTION
**Pull request by Dhaiwat10**
- Closes #2995 

# Release notes

In this release, we:

- Added an embedded testnet faucet into the `create-fuels` template app

# Summary

This PR lets users use the testnet faucet via an embedded iframe from the template app, without having to leave the app just to get faucet funds.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: templates/nextjs/src/components/Layout.tsx [hunk 1]
```diff
@@ -9,9 +9,11 @@ import { useActiveWallet } from "@/hooks/useActiveWallet";
 import { useFaucet } from "@/hooks/useFaucet";
 import Head from "next/head";
 import { bn } from "fuels";
+import { useRouter } from "next/router";
 
 export const Layout = ({ children }: { children: React.ReactNode }) => {
   const { faucetWallet } = useFaucet();
+  const router = useRouter();
```
Comment (petertonysmith94) [inline templates/nextjs/src/components/Layout.tsx:16]: Receiving the following error when using this template:

> Error: NextRouter was not mounted. https://nextjs.org/docs/messages/next-router-not-mounted

image

File: templates/nextjs/src/app/faucet/page.tsx [hunk 2]
```diff
@@ -50,36 +51,54 @@ export default function Faucet() {
...
+      {CURRENT_ENVIRONMENT === Environments.TESTNET && (
+        <>
+          <iframe
+            src={
+              wallet
+                ? `${TESTNET_FAUCET_LINK}?address=${wallet.address.toB256()}`
+                : TESTNET_FAUCET_LINK
+            }
+            title="faucet"
+            className="w-full h-screen ovreflow-scroll"
```
Comment (petertonysmith94) [inline templates/nextjs/src/app/faucet/page.tsx:98]:
```suggestion
            className="w-full h-screen overflow-scroll"
```

File: templates/vite/src/routes/faucet.lazy.tsx [hunk 3]
```diff
@@ -53,36 +54,54 @@ function Index() {
...
+      {CURRENT_ENVIRONMENT === Environments.TESTNET && (
+        <>
+          <iframe
+            src={
+              wallet
+                ? `${TESTNET_FAUCET_LINK}?address=${wallet.address.toB256()}`
+                : TESTNET_FAUCET_LINK
+            }
+            title="faucet"
+            className="w-full h-screen ovreflow-scroll"
```
Comment (petertonysmith94) [inline templates/vite/src/routes/faucet.lazy.tsx:101]:
```suggestion
            className="w-full h-screen overflow-scroll"
```

**Review by petertonysmith94 — COMMENTED** [review]
Will this affect the UI tests?

**Comment by Dhaiwat10** [comment]
@petertonysmith94 this shouldn't affect the UI integration tests as I believe those are run in the local environment whereas the testnet faucet only shows up in the testnet environment

**Review by petertonysmith94 — APPROVED** [review]

**Review by maschad — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Pull request merged**